juju/juju
# Merge 2.9

Merge 2.9

#14166 [JUJU-1331] Extract embedded func to new loadOneWatcherEntity.
#14189 [JUJU-1355] Add jammy template data for vsphere provider
#14132 [JUJU-1266] Rename column DNS --> Address in machines tabular output
#14121 [JUJU-1246] add deprecation notice to the 'juju sla' command
#14192 Move RetryProvisioning and ProvisioningScript methods off client facade
#14194 Move constraints and add-charm apis off client facade

Conflicts were due to removal of legacy code for 3.0
```
Conflicts:
#       api/client/charms/client.go
#       api/client/charms/client_test.go
#       api/client/client/client_test.go
#       apiserver/facades/client/client/client.go
#       apiserver/facades/client/client/client_test.go
#       apiserver/facades/client/machinemanager/machinemanager.go
#       apiserver/facades/client/machinemanager/machinemanager_test.go
#       apiserver/facades/client/machinemanager/register.go
#       apiserver/facades/client/modelconfig/modelconfig.go
#       apiserver/facades/client/modelconfig/register.go
#       apiserver/facades/schema.json
#       cmd/juju/application/deploy.go
#       cmd/juju/application/refresh.go
#       cmd/juju/application/store/interface.go
#       cmd/juju/machine/add.go
#       cmd/juju/machine/add_test.go
#       cmd/juju/machine/remove.go
```

## QA steps

See PRs

[JUJU-1331]: https://warthogs.atlassian.net/browse/JUJU-1331?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-1355]: https://warthogs.atlassian.net/browse/JUJU-1355?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-1266]: https://warthogs.atlassian.net/browse/JUJU-1266?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-1246]: https://warthogs.atlassian.net/browse/JUJU-1246?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
